unitybuild.lua

```diff
@@ -58,26 +120,50 @@ function premake.extensions.compilationunit.customBakeFiles(base, prj)
 	-- enabled on the command line with the `--unity-build` option.
 	local unity_build_enabled = prj.unitybuildenabled and (_OPTIONS['unity-build'] ~= nil)
 	local cu = premake.extensions.compilationunit
+	local project_name = prj['name']
 
 	-- do nothing for external projects
 	if prj.external == true then
 		return base(prj)
 	end
 
+	-- make sure this local var stays a boolean in all cases.
+	if unity_build_enabled == nil then
+		unity_build_enabled = false
+	end
```

Comment:
is it not a boolean? The way it's written above seems like it should be always boolean. Or you could be more explicit above with:
```
	local unity_build_enabled = (prj.unitybuildenabled and (_OPTIONS['unity-build'] ~= nil)) or false
```

Reply:
it can be `nil` in some cases apparently.  It has to do with how the condition above evaluates.  I was hitting specifically this case where the local variable came out to be `nil` in some situations in projects that didn't opt-in to unity builds.